eth-contracts/contracts/service/storage/ServiceProviderStorage.sol

```diff
@@ -37,22 +37,12 @@ contract ServiceProviderStorage is RegistryContract {
     /** @dev - mapping of address -> sp id array */
     /** @notice - stores all the services registered by a provider. for each address,
     provides the ability to lookup by service type and see all registered services */
-    mapping(address => mapping(bytes32 => uint)) serviceProviderAddressToId;
+    mapping(address => mapping(bytes32 => uint[])) serviceProviderAddressToId;
```

Comment:
for this storage contract, we just pass in ownerAddress everywhere and do no validation. i wonder if it's possible to call functions on behalf of others?

Reply:
we can definitely add some validation here - let me give this a once over to confirm

Reply:
ah so the reason here is bc msg.sender from the factory <-> storage is always the same address - we pass in owner address to actually associate the endpoints with the caller of sp factory. there is lots of validation in terms of confirming the owner though, not sure what you meant there? for example you cannot deregister an endpoint without the correct owner address, or update the delegate owner wallet without the right owner address. there are tests confirming this behavior as well.

Reply:
i will confirm that storage is not publicly accessible